### ISSUE TITLE
Webpack 4 support

### ISSUE THREAD

**Issue by filipesilva** [postcss/postcss-loader#319]
Briefly describe the issue you are experiencing (or the feature you want to see
added to the plugin). Tell us what you were trying to do and what happened
instead. Remember, this is _not_ a place to ask questions. For that, go to
http://gitter.im/postcss/postcss

### `Details`
---

Trying to use `postcss-loader` with `webpack@4.0.0-alpha.0` (https://github.com/webpack/webpack/issues/6064).

### `Error (Logs|Stacks)`
---

The main error seems to be `Module build failed: TypeError: loader._compilation.applyPluginsWaterfall is not a function`.

```
ERROR in D:/work/angular/aio/node_modules/css-loader?{"sourceMap":false,"importLoaders":1}!D:/work/angular/aio/node_modules/postcss-loader?{"ident":"postcss","sourceMap":false}!D:/work/angular/aio/node_modules/sass-loader/lib/loader.js?{"sourceMap":false,"precision":8,"includePaths":[]}!D:/work/angular/aio/src/styles.scss
Module build failed: TypeError: loader._compilation.applyPluginsWaterfall is not a function
    at D:\work\angular\aio\node_modules\postcss-loader\index.js:122:43
    at <anonymous>
 @ D:/work/angular/aio/src/styles.scss 4:14-189
 @ multi D:/work/angular/aio/src/styles.scss
```

### `Reproduction [Code]`
---

I think this happens whenever `postcss-loader` is used with Webpack 4, but if you think that's not the case and need to zone in on the issue let me know and I will make a repro.

### `Environment`
---

Please provide information about your environment.

|OS|node|npm|PostCSS|
|:-:|:--:|:-:|:------:|
|Windows 10|8.9.1|5.6.0|2.0.8|

**Comment by michael-ciniawsky** [comment]
hmm.. I wonder where this is actually coming from in `v2.0.0` since I removed the 'Custom Plugin Compilation Hook' in `v1.0.0...v2.0.0`. Could you point my to the where in `angular-cli` the loader is getting used atm ?

**Comment by filipesilva** [comment]
We use it here: https://github.com/angular/angular-cli/blob/master/packages/@angular/cli/models/webpack-configs/styles.ts#L175-L182

But now that you mention the versions, I went and checked and it seems that in that project the `postcss-loader` version was actually 1.3.x. I updated to 2.0.8 and didn't have that problem anymore. So, user error :/

**Comment by TimeBandit** [comment]
Saw same issue, upgrading to `"postcss-loader": "^2.1.5"` and the problem went away

**Comment by OliverJAsh** [comment]
What was the exact fix for this issue, and in which release did it first appear?

Upgrading to the latest version isn't a great solution, as many times that only replaces old bugs for new bugs. If we can work in terms of specific fixes, then we can deal with bugs on a case-by-case basis.

**Comment by OliverJAsh** [comment]
It appears the fix for this issue first appeared in v2.0.0, although the changelog doesn't mention it: https://github.com/postcss/postcss-loader/blob/928d5c41d5e2bdec130b3c0899760889466ae7bd/CHANGELOG.md#200-2017-05-08
